### PR TITLE
Add `reason` when cast vote

### PR DESCRIPTION
**Pull request by twodam**
Fixes Issue #3147

Changes in this PR:
* submit vote with reason
* query votes with reason
* display vote reason if not empty

How to test and review this PR?
you can vote on any proposal with reason and then check the result.

Screenshot:
image
image

Potential related issue:
it seems the API didn't return newest reason after user change their vote option. (newest option but not newest reason).

### CODE REVIEW

**Comment by bonustrack** [comment]
Thanks for your contribution @twodam, here is some feedback:

- The input currently look like this:
image
But this doesn't follow the style that we have on others inputs, we should change to a style like this (taken from space settings page):
image
As you can see in this example the background is different, padding too, there isn't any outline too.

- Here we should use the icon `annotation` from Heroicons and not the `information-circle` icon, also the size should be the same as the sign icon next to it.
image

For the issue on the hub it will be fixed on next update.

**Comment by twodam** [comment]
> Thanks for your contribution @twodam, here is some feedback:
> 
> * The input currently look like this:
> 
> image
> 
> But this doesn't follow the style that we have on others inputs, we should change to a style like this (taken from space settings page):
> image
> 
> As you can see in this example the background is different, padding too, there isn't any outline too.
> * Here we should use the icon `annotation` from Heroicons and not the `information-circle` icon, also the size should be the same as the sign icon next to it.
> 
> image
> 
> For the issue on the hub it will be fixed on next update.

how can I add icon `annotation`? do I need to upload things in `src/assets/fonts`?

**Comment by twodam** [comment]
I found five different style for `TextareaAutosize`, what do you want?

1. `class="s-input !rounded-3xl"`
2. `class="input w-full text-left"`
3. `class="no-scrollbar w-full !overflow-x-scroll whitespace-pre font-mono text-sm"`
4. `class="input w-full text-left"`
5. `class="input h-full w-full text-left"`

File: src/components/SpaceProposalVotesList.vue [hunk 1]
```diff
@@ -151,6 +151,15 @@ watch(visibleVotes, () => {
...
           <BaseIcon name="signature" />
         </BaseButtonIcon>
+        <span
+          v-if="vote.reason !== ''"
+          v-tippy="{
+            content: vote.reason
+          }"
+          class="text-skin-text transition-colors duration-200 hover:text-skin-link"
+        >
+          <BaseIcon name="info" />
```
Comment (samuveth) [inline src/components/SpaceProposalVotesList.vue:161]: Hi @twodam, you don't need to add it, all hero icons are already installed and can be used like this.
```
<i-ho-annotation />
```
To change the size you can use tailwind css text size.

The `<BaseIcon>` is how we did it before and we haven't fully migrated all icons yet.

Comment (samuveth) [reply src/components/SpaceProposalVotesList.vue:161]: Here is a full list of the installed hero icons https://icon-sets.iconify.design/heroicons-outline/

File: src/components/ModalConfirm.vue [hunk 2]
```diff
@@ -145,6 +147,12 @@ watch(
             <BaseIcon name="info" size="24" class="text-skin-text" />
           </BaseLink>
         </div>
+        <div class="flex">
+          <span class="mr-1 flex-auto text-skin-text" v-text="$t('reason')" />
+        </div>
+        <div class="flex">
+          <TextareaAutosize v-model="reason" :max-length="140" />
```
Comment (samuveth) [inline src/components/ModalConfirm.vue:154]: We should have `class="s-input !rounded-3xl"` on TextareaAutosize. I see you found many different stylings for Textareas, these are mostly left overs that need to be refactored

File: src/locales/zh-CN.json [hunk 3]
```diff
@@ -67,6 +67,7 @@
   "quorumReached": "已达到既定人数",
   "options": "选项",
   "votingPower": "您的投票权",
+  "reason": "理由",
```
Comment (samuveth) [inline src/locales/zh-CN.json:70]: Translations should not be part of this PR

Comment (samuveth) [inline src/locales/zh-CN.json:70]:
```suggestion
```

**Comment by twodam** [comment]
@samuveth @bonustrack thanks for detailed feedback, I updated those place, can you review again?

File: src/components/ModalConfirm.vue [hunk 4]
```diff
@@ -87,9 +87,9 @@ watch(
   >
     <div class="flex flex-auto flex-col">
       <h4 class="m-4 mb-0 text-center">
-        {{ $tc('proposal.castVote') }}
+        {{ $tc('voteOverview') }}
```
Comment (samuveth) [inline src/components/ModalConfirm.vue:90]: @twodam why were these changes reverted?

Comment (twodam) [reply src/components/ModalConfirm.vue:90]: I believe it's because base branch are updated and you commit once before to manually `merge` them.
Anyway, check `Files changed` section, it should no longer touch these lines.

File: src/locales/default.json [hunk 5]
```diff
@@ -67,6 +67,10 @@
   "quorumReached": "quorum reached",
   "options": "Option(s)",
   "votingPower": "Your voting power",
+  "reason": {
+    "label": "Reason",
+    "placeholder": "What is your reason?"
```
Comment (samuveth) [inline src/locales/default.json:72]: I think this is to aggressive way to ask the user. Instead I would suggest

`Leave a comment (optional)`

File: src/locales/default.json [hunk 6]
```diff
@@ -67,6 +67,10 @@
   "quorumReached": "quorum reached",
   "options": "Option(s)",
   "votingPower": "Your voting power",
+  "reason": {
+    "label": "Reason",
```
Comment (samuveth) [inline src/locales/default.json:71]: `Reason` as label seems to limiting. Instead I would call it `Comment`

Comment (twodam) [reply src/locales/default.json:71]: Won't this confuse user with comment plugin? And the name `reason` are referenced in `snapshot.js vote()`(same as field in wallet signing UI)

Comment (samuveth) [reply src/locales/default.json:71]: No I think it is clear the user is leaving the comment on their vote. I will discuss further with Fabien about this

File: src/components/ModalConfirm.vue (hunk 4 above)
Comment (samuveth) [reply src/components/ModalConfirm.vue:90]: I still don't see these changes I pushed to this PR. Can you add them back

Comment (samuveth) [reply src/components/ModalConfirm.vue:90]: I would like this changes to be included that were removed here https://github.com/snapshot-labs/snapshot/commit/9ce757ac9505ce991377554b0e3b593e704b1422

**Comment by samuveth** [comment]
image
The arrow is off-center

**Comment by samuveth** [comment]
@twodam I've fixed my remaining issues and I've given it to final review.
